packages/actor-init-sparql/test/ActorInitSparql-test.ts

```diff
@@ -621,6 +621,24 @@ describe('ActorInitSparql', () => {
             '@comunica/actor-init-sparql:baseIRI': 'myBaseIRI',
           });
       });
+
+      it('should allow optimize actors to modify the context', async() => {
+        const optimizeMediate = mediatorOptimizeQueryOperation.mediate;
+        mediatorOptimizeQueryOperation.mediate = (action: any) => {
+          action = {
+            ...action,
+            context: action.context.set('the-answer', 42),
+          };
+          return optimizeMediate(action);
+        };
+        const queryMediate = mediatorQueryOperation.mediate;
+        mediatorQueryOperation.mediate = (action: any) => {
+          expect(action.context.get('the-answer')).toBe(42);
```

Comment:
For test safety, I would move this `expect` to after `actor.query`. Because if this method is not called, then this assertion will never be tested, and the test may not show up as a failure.

Reply:
Done! And while doing this, I realized I could rewrite the test in a much simpler way.